src/tfr_control/src/arm_action_server.cpp

```diff
@@ -6,77 +6,93 @@
-
-    // Try to plan to the given target
-    bool success = (move_group->plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
-
-    if (success) {
-        // Planning was successful, actually execute the movement
-        ROS_DEBUG("Executing movement");
-        move_group->move();
-        // TODO: How are errors handled?
-        //  - Further note: in the case that, say, the arm gets stuck and can't
-        //    move, how are these errors returned? Exceptions, return code, or
-        //    something else?
-    } else {
-        ROS_WARN("Planning of movement failed, not executing");
+class ArmActionServer {
+public:
+    ArmActionServer(ros::NodeHandle &n) : move_group{"arm_end"}, joint_model_group(*move_group.getCurrentState()->getJointModelGroup("arm_end")),
+        server{n, "move_arm", boost::bind(&ArmActionServer::execute, this, _1), false} {
+        server.start();
     }
 
-    // Get the current position to return in the result message
-    std::vector<double> current_pos;
-    move_group->getCurrentState()->copyJointGroupPositions(joint_model_group, current_pos);
-
-    // Populate the result message
-    tfr_msgs::ArmMoveResult result;
-    result.pose.resize(4);
-    result.pose[0] = current_pos[0];
-    result.pose[1] = current_pos[1];
-    result.pose[2] = current_pos[2];
-    result.pose[3] = current_pos[3];
-
-    // Send the result message and set the appropriate action server status
-    if (success) {
-        server->setSucceeded(result);
-    } else {
-        server->setAborted(result);
+private:
+    // This is the method that will be called when a client makes use
+    // of this server. The provided goal is the "input"
+    void execute(const tfr_msgs::ArmMoveGoalConstPtr& goal)
+    {
+        // Set up the joint space goal vector to travel to based on the input goal
+        // from the action server
+        std::vector<double> joint_group_positions(4);
+        joint_group_positions[0] = goal->pose[0];
+        joint_group_positions[1] = goal->pose[1];
+        joint_group_positions[2] = goal->pose[2];
+        joint_group_positions[3] = goal->pose[3];
+
+        // Set the current target
+        move_group.setJointValueTarget(joint_group_positions);
+        moveit::planning_interface::MoveGroupInterface::Plan my_plan;
+
+        // Try to plan to the given target
+        bool success = (move_group.plan(my_plan) == MoveItErrorCode::SUCCESS);
+        bool moveSuccess = true; // Defaults to true
+
+        if (success) {
+            // Planning was successful, actually execute the movement
+            ROS_DEBUG("Executing movement");
+            moveSuccess = (move_group.move() == MoveItErrorCode::SUCCESS);
+
+            if (!success) {
+                ROS_WARN("Planning sucessful, but execution failed");
+            }
+        } else {
+            ROS_WARN("Planning of movement failed, not executing");
+        }
```

Comment:
Current coding standard says braces get their own line.

Reply:
Thanks. I'm pretty sure I did this on accident in a few other places, so I'll quickly scan all of my files to be sure (already caught one in my constructor as well).